src/result_object/index.js

```diff
@@ -62,10 +61,29 @@ class ResultObject {
      * @param {string} fieldName - The name of the field.
      * @param {string} statement - The error string to add to the object.
      * @param {string} severity - Whether it is a `fail` or `warn` test.
-     * @return {Function} Calling a helper function
+     * @return {Object} Current instance
      */
-    fail(fieldName: string, statement: string, severity: Severity = FAIL) {
-        return fail.call(this, fieldName, statement, severity);
+    fail(fieldName: string, statement: string, severity: Severity) {
+        const selectedSeverity: Severity = severity && severities.includes(severity) ? severity : FAIL;
+        const isWarn: boolean = selectedSeverity === WARN;
+
+        const countName: string = isWarn ? 'warnCount' : 'failCount';
+
+        if (isWarn) {
+            this.hasValidationWarnings = true;
+            this.validationWarnings[fieldName] = this.validationWarnings[fieldName] || [];
+            this.validationWarnings[fieldName].push(statement);
```

Comment:
I think it will be clearer to juxtapose 
```js
this.testsPerformed[fieldName].warnCount++;
this.warnCount++;
```

or maybe you can create a function for each of them: `bumpTestCounter`, `bumpWarnCounter`, `bumpFailCounter`